hbase_sql/src/main/java/com/splicemachine/olap/AsyncOlapNIOLayer.java

```diff
@@ -95,6 +102,23 @@ private void connect() throws IOException {
             HostAndPort hap = hostProvider.olapServerHost(queue);
             LOG.info("Connecting to " + hap);
 
+            try{
+                String root = HConfiguration.getConfiguration().getSpliceRootPath();
+                String queueRoot = root + HBaseConfiguration.OLAP_SERVER_PATH + HBaseConfiguration.OLAP_SERVER_QUEUE_PATH;
+                ZkUtils.getChildren(queueRoot, new Watcher() {
+                    @Override
+                    public void process(WatchedEvent watchedEvent) {
+                        if (watchedEvent.getType().equals(Event.EventType.NodeChildrenChanged)) {
+                            SpliceLogUtils.warn(LOG, "A new OLAP server was launched. Disconnecting...");
+                            connected = false;
+                        }
+                    }
+                });
+            } catch (IOException e) {
+                // This could happen for UT when no zookeeper quorum is available
+                SpliceLogUtils.warn(LOG, "Zookeeper is not available", e);
+            }
+
```

Comment:
I don't like this strong dependency on ZK here, this logic could be better encapsulated so that this class doesn't require ZK (such as what we do with OlapServerProvider). But I'm ok with merging it for now.

Reply:
Should this be Jira'd as tech debt?